From e323de473d292a5c7f21e6c6c493bc94aab62739 Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Mon, 3 Jun 2024 15:20:57 +0200
Subject: media: atomisp: csi2-bridge: Add DMI quirk for OV5693 on Xiaomi
 Mipad2

The OV5693 the Xiaomi Mipad2 is used as a front facing sensor and
as such is connected to CsiPort 0, but the _DSM has CsiPort 1, add
a DMI quirk to override the wrong _DSM setting.

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
Reviewed-by: Andy Shevchenko <andy@kernel.org>
Link: https://lore.kernel.org/r/20240603132057.255917-5-hdegoede@redhat.com
Signed-off-by: Hans Verkuil <hverkuil-cisco@xs4all.nl>
---
 drivers/staging/media/atomisp/pci/atomisp_csi2_bridge.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/drivers/staging/media/atomisp/pci/atomisp_csi2_bridge.c b/drivers/staging/media/atomisp/pci/atomisp_csi2_bridge.c
index 2483eaeeac7332..d789d38ef68993 100644
--- a/drivers/staging/media/atomisp/pci/atomisp_csi2_bridge.c
+++ b/drivers/staging/media/atomisp/pci/atomisp_csi2_bridge.c
@@ -106,6 +106,12 @@ static struct gmin_cfg_var lenovo_ideapad_miix_310_vars[] = {
 	{}
 };
 
+static struct gmin_cfg_var xiaomi_mipad2_vars[] = {
+	/* _DSM contains the wrong CsiPort for the front facing OV5693 sensor */
+	{ "INT33BE:00", "CsiPort", "0" },
+	{}
+};
+
 static const struct dmi_system_id gmin_cfg_dmi_overrides[] = {
 	{
 		/* Lenovo Ideapad Miix 310 */
@@ -115,6 +121,14 @@ static const struct dmi_system_id gmin_cfg_dmi_overrides[] = {
 		},
 		.driver_data = lenovo_ideapad_miix_310_vars,
 	},
+	{
+		/* Xiaomi Mipad2 */
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Xiaomi Inc"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Mipad2"),
+		},
+		.driver_data = xiaomi_mipad2_vars,
+	},
 	{}
 };
 
-- 
cgit 1.2.3-korg
